discovery/consul_test.go

```diff
@@ -77,13 +77,18 @@ a Consul server for testing. The 'consul' binary must be in the $PATH
 ref https://github.com/hashicorp/consul/tree/master/testutil
 */
 
-var testServer *testutil.TestServer
+var testServer *TestServer
 
 func TestWithConsul(t *testing.T) {
-	testServer, _ = testutil.NewTestServerConfigT(t, func(c *testutil.TestServerConfig) {
-		c.LogLevel = "err"
-	})
+	var err error
+	testServer, err = NewTestServer(8500)
+	if err != nil {
+		t.Fatal(err)
+	}
 	defer testServer.Stop()
+
+	time.Sleep(300 * time.Millisecond)
```

Comment:
I'm not crazy about this and would much rather add something that checks for Consul's HTTP API endpoint being live. Any suggestions on that?

Reply:
Query `localhost:8500/v1/status/leader` for a non-empty `200 OK` response but do that as part of the `Start()` call. You can use/clone the `testutil/retry` package for that.

Reply:
Our tests don't require leadership election so I was able to just throw a naive GET request at `/v1/agent/self` which also uses `testutil/retry`.